src/core/Makefile.am

```diff
@@ -3,23 +3,22 @@ libdyad_core_la_SOURCES = \
 	dyad_core.c
 libdyad_core_la_LIBADD = \
 	$(top_builddir)/src/dtl/libdyad_dtl.la \
-	$(UCX_LIBS) \
 	$(JANSSON_LIBS) \
 	$(FLUX_CORE_LIBS)
 libdyad_core_la_CFLAGS = \
 	$(AM_CFLAGS) \
 	-I$(top_srcdir)/src/utils \
 	-I$(top_srcdir)/src/utils/base64 \
 	-I$(top_srcdir)/src/dtl \
-    $(UCX_CFLAGS) \
     $(JANSSON_CFLAGS) \
 	$(FLUX_CORE_CFLAGS) \
 	-DBUILDING_DYAD=1 \
 	-fvisibility=hidden
 libdyad_core_la_CPPFLAGS = 
-libdyad_core_la_LDFLAGS = \
-	-Wl,-rpath,'$(UCX_LIBDIR)' \
```

Comment:
do u wanna add UCX_LIBDIR? we have to make sure its pkgconfig correctly sets these.

What about rpath? skipping this will make libs dynamically loaded based on LD_LIBRARY_PATH.

Reply:
I've already checked that UCX's pkgconfig sets `libdir`. It works as expected.

As for your question about `rpath`, I have removed that flag from `libdyad_core.so`. 

The `rpath` flag is now only set for the module (i.e., `dyad.so`). This seemingly has to be done for the module due to some oddities when launching it with `flux module load`. For some reason, I've found that no environment variables from the user's environment are carried over into the launch of the module. So, `LD_LIBRARY_PATH` does nothing to the module. This is a problem for us because UCX has to be loaded for the module to work. By setting `rpath` for the module, I can get around this oddity.

Reply:
Thats fine.